src/borrow_tracker/tree_borrows/perms.rs

```diff
@@ -87,6 +87,28 @@ impl PermissionPriv {
     fn compatible_with_protector(&self) -> bool {
         !matches!(self, ReservedIM)
     }
+
+    /// Returns the strongest foreign action this node survives (without change),
+    /// where `prot` indicates if it is protected.
+    /// They are ordered as `None` < `Some(Read)` < `Some(Write)`, in order of power.
+    pub fn strongest_survivable_foreign_action(&self, prot: bool) -> Option<AccessKind> {
```

Comment:
This function can't actually be invoked from the outside since `PermissionPriv` is private, I think? So please remove the `pub`.